source/agora/node/Validator.d

```diff
@@ -83,6 +86,9 @@ public class Validator : FullNode, API
             this.network.discover(required_peer_keys);
             this.network.startPeriodicCatchup(this.ledger, &this.nominator.isNominating);
         });
+
+        // start a periodic timer that `checkRevealPreimage` the time 10 seconds
+        this.taskman.timer(10.seconds, toDelegate(&checkRevealPreimage));
```

Comment:
Why do you need `toDelegate`? `&checkRevealPreimage` is already a delegate.

Reply:
I remove it.